web/src/scheduled_messages.js

```diff
@@ -0,0 +1,158 @@
+}
+
+export function edit_scheduled_message(scheduled_msg_id) {
+    const scheduled_msg = scheduled_messages_data.find(
+        (msg) => msg.message_id === scheduled_msg_id,
+    );
+
+    let compose_args;
+
+    if (scheduled_msg.type === "stream") {
+        compose_args = {
+            type: "stream",
+            stream: scheduled_msg.stream_name,
+            topic: scheduled_msg.topic,
+            content: scheduled_msg.content,
+        };
+    } else {
+        const recipient_emails = [];
+        if (scheduled_msg.to) {
+            for (const recipient_id of scheduled_msg.to) {
+                recipient_emails.push(people.get_by_user_id(recipient_id).email);
+            }
+        }
+        compose_args = {
+            type: scheduled_msg.type,
+            private_message_recipient: recipient_emails.join(","),
+            content: scheduled_msg.content,
+        };
+    }
+
+    if (compose_args.type === "stream") {
+        narrow.activate(
+            [
+                {operator: "stream", operand: compose_args.stream},
+                {operator: "topic", operand: compose_args.topic},
+            ],
+            {trigger: "edit scheduled message"},
+        );
+    } else {
+        narrow.activate([{operator: "pm-with", operand: compose_args.private_message_recipient}], {
+            trigger: "edit scheduled message",
+        });
+    }
+
+    overlays.close_overlay("scheduled");
+    compose_fade.clear_compose();
+    compose.clear_preview_area();
+    compose_actions.start(compose_args.type, compose_args);
+    compose_ui.autosize_textarea($("#compose-textarea"));
+    $("#compose-textarea").attr("data-scheduled-message-id", scheduled_msg_id);
+}
+
+function format(scheduled_messages) {
+    for (const msg of scheduled_messages) {
+        if (msg.type === "stream") {
+            msg.is_stream = true;
+            msg.stream_id = msg.to[0];
+            msg.stream_name = stream_data.maybe_get_stream_name(msg.stream_id);
+            const color = stream_data.get_color(msg.stream_name);
+            msg.recipient_bar_color = stream_color.get_recipient_bar_color(color);
+            msg.stream_privacy_icon_color = stream_color.get_stream_privacy_icon_color(color);
+        } else {
+            msg.is_stream = false;
+            msg.recipients = people.get_recipients(msg.to.join(","));
+        }
+        const time = new Date(msg.deliver_at);
+        msg.time_stamp = timerender.time_left_to_achieve_date(time);
+        msg.full_date_time = timerender.get_full_datetime(time);
+    }
+}
+
+export function delete_scheduled_message(scheduled_msg_id) {
+    channel.del({
+        url: "/json/scheduled_messages/" + scheduled_msg_id,
+        success() {
+            if (overlays.scheduled_messages_open()) {
+                $(
+                    `#scheduled_messages_overlay .scheduled-message-row[data-message-id=${scheduled_msg_id}]`,
+                ).remove();
+            }
```

Comment:
This is probably OK for now, but we should have a comment that this live-update code is here rather than `server_events_dispatch.js` because we don't have server events for this part of the system.

Reply:
Done.